iml-agent/src/action_plugins/lustre/snapshot/parse.rs

```diff
@@ -0,0 +1,310 @@
+// Copyright (c) 2020 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use chrono::offset::Utc;
+use chrono::DateTime;
+use combine::{
+    attempt, choice, eof,
+    error::{ParseError, StreamError},
+    many1,
+    parser::{char, repeat::take_until},
+    satisfy, skip_many,
+    stream::{Stream, StreamErrorFor},
+    Parser,
+};
+use iml_wire_types::snapshot::{Detail, Snapshot, Status};
+use std::collections::HashMap;
+
+#[derive(PartialEq, Debug)]
+enum Segment {
+    Snapshot(Snapshot),
+    Detail(Detail),
+}
+
+fn spaces<I>() -> impl Parser<I, Output = ()>
```

Comment:
Could we use https://docs.rs/combine/4.2.1/combine/parser/char/fn.spaces.html instead?

Reply:
Here no, because it should not include new lines (`combine::parser::char::spaces` does).